No location for 'must return a value' error

**TypeScript Version:**  3.9.0-dev.20200305

**Code**

```js
/** @type {function(): number} */
function f() {}
```

**Expected behavior:**
```
a.js:1:24 - error TS2355: A function whose declared type is neither 'void' nor 'any' must return a value.
```

**Actual behavior:**
```
error TS2355: A function whose declared type is neither 'void' nor 'any' must return a value.
```

--- a/src/compiler/utilitiesPublic.ts
+++ b/src/compiler/utilitiesPublic.ts
@@ -769,7 +769,7 @@ namespace ts {
                 const sig = find(type.members, isCallSignatureDeclaration);
                 return sig && sig.type;
             }
-            if (isFunctionTypeNode(type)) {
+            if (isFunctionTypeNode(type) || isJSDocFunctionType(type)) {
                 return type.type;
             }
         }

--- a/tests/baselines/reference/asyncArrowFunction_allowJs.errors.txt
+++ b/tests/baselines/reference/asyncArrowFunction_allowJs.errors.txt
@@ -1,11 +1,13 @@
 tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(3,17): error TS2322: Type '0' is not assignable to type 'string'.
+tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(6,24): error TS1064: The return type of an async function or method must be the global Promise<T> type.
 tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(7,23): error TS2322: Type '0' is not assignable to type 'string'.
+tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(10,24): error TS1064: The return type of an async function or method must be the global Promise<T> type.
 tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(12,2): error TS2322: Type '0' is not assignable to type 'string'.
 tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(19,3): error TS2345: Argument of type '() => Promise<number>' is not assignable to parameter of type '() => string'.
   Type 'Promise<number>' is not assignable to type 'string'.
 
 
-==== tests/cases/conformance/async/es2017/asyncArrowFunction/file.js (4 errors) ====
+==== tests/cases/conformance/async/es2017/asyncArrowFunction/file.js (6 errors) ====
     // Error (good)
     /** @type {function(): string} */
     const a = () => 0
@@ -14,12 +16,16 @@ tests/cases/conformance/async/es2017/asyncArrowFunction/file.js(19,3): error TS2
     
     // Error (good)
     /** @type {function(): string} */
+                           ~~~~~~
+!!! error TS1064: The return type of an async function or method must be the global Promise<T> type.
     const b = async () => 0
                           ~
 !!! error TS2322: Type '0' is not assignable to type 'string'.
     
     // No error (bad)
     /** @type {function(): string} */
+                           ~~~~~~
+!!! error TS1064: The return type of an async function or method must be the global Promise<T> type.
     const c = async () => {
     	return 0
     	~~~~~~~~

--- a/tests/baselines/reference/jsdocFunction_missingReturn.errors.txt
+++ b/tests/baselines/reference/jsdocFunction_missingReturn.errors.txt
@@ -1,0 +1,9 @@
+/a.js(1,24): error TS2355: A function whose declared type is neither 'void' nor 'any' must return a value.
+
+
+==== /a.js (1 errors) ====
+    /** @type {function(): number} */
+                           ~~~~~~
+!!! error TS2355: A function whose declared type is neither 'void' nor 'any' must return a value.
+    function f() {}
+    

--- a/tests/baselines/reference/jsdocFunction_missingReturn.symbols
+++ b/tests/baselines/reference/jsdocFunction_missingReturn.symbols
@@ -1,0 +1,5 @@
+=== /a.js ===
+/** @type {function(): number} */
+function f() {}
+>f : Symbol(f, Decl(a.js, 0, 0))
+

--- a/tests/baselines/reference/jsdocFunction_missingReturn.types
+++ b/tests/baselines/reference/jsdocFunction_missingReturn.types
@@ -1,0 +1,5 @@
+=== /a.js ===
+/** @type {function(): number} */
+function f() {}
+>f : () => number
+

--- a/tests/cases/conformance/jsdoc/jsdocFunction_missingReturn.ts
+++ b/tests/cases/conformance/jsdoc/jsdocFunction_missingReturn.ts
@@ -1,0 +1,8 @@
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+// @strict: true
+
+// @Filename: /a.js
+/** @type {function(): number} */
+function f() {}

